docs/DeveloperGuide.adoc

```diff
@@ -1,93 +1,1101 @@
+
+* **Alternative 2:** Each command knows what access level is required to run it
+** Pros: Easy to find what access level a command requires
+** Cons: Difficult to see what commands a given User can run. Difficult to organize the commands by their required access level.
+
+* **Alternative 3:** Have both 1 and 2
+** Pros: Doing both the above operation becomes easy
+** Cons: Difficult to maintain both information. Access level information is now stored in 2 separate locations and needs to be in sync
+
+===== Aspect: How to update the privilege of Logic given an Account to log into
+
+* **Alternative 1 (current choice):** The Privilege object attached to Logic will copy the User and the Person attached to the given Account object, into its own user and myPerson respectively.
+** Pros: Does not require the Command to interact with Logic, thus reducing coupling.
+** Cons: A bit more awkward solution than the Alternative 2
+* **Alternative 2:** Change Logic to point to the new Privilege object inside of Account.
+** Pros: Quick to implement.
+** Cons: Higher level of coupling compared to Alternative 1. Also, would require storing or instantiating a default Privilege and making Logic point to this new object if Logout Command were to be called. 
+
+===== Aspect: Default required access levels for commands
+
+* **Alternative 1 (current choice):** Access level is set to Admin by default
+** Pros: If a developer forgets to give access level to a Command, it will to still be functional.
+** Cons: May mask the fact that the Command have the unintended access level of Admin.
+
+* **Alternative 2:** Enforce all Commands to be assigned a access level
+** Pros: Forces delevopers to be deliberate with their access level assignment to commands
+** Cons: Program will fail to integrate if any of the developer forgot to assign an access level to their newly developed commands.
+
+===== Aspect: Who throws InsufficientPrivilege Exception
+
+* **Alternative 1 (current choice):** Logic throws the Exception
+** Pros: It is a common starting point of commands, so it is easy to implement it to throw the Exception
+** Cons: Blurs the responsiblity of commands being the one throwing the Exception
+
+* **Alternative 2:** Each Command checks the Privilege level during execute()
+** Pros: Keeps to the responsibility of Commands being the one to throw the Exception when they are being executed
+** Cons: Requires backdating all the commands to include this function in their execute(). Also prone to errors as a developer may forget to call this function in their execute() of their newly developed Commands.
+
+* **Alternative 3:** Have a function that checks the Privilege level in parent Command that is called during execute()
+** Pros: Same as 2.
+** Cons: Same as 2, but instead of using individual functions, we are repeatedly using the inherited function instead
+// end::privilege[]
+
+// tag::Grades[]
+=== Grades Feature
```

Comment:
It would be nice to include an activity diagram to describe your solution workflow.